### PR TITLE
fix: unnecessary background refresh on query page

### PR DESCRIPTION
**Pull request by astandrik**
[Stand](http://ydb-vla-dev02-005.search.yandex.net:8765/monitoring/tenant?tenantPage=query&diagnosticsTab=topQueries&topQueriesDateFrom=1721228339103&topQueriesDateTo=1721314739103&selectedConsumer=&name=%2Fdev02&clckid=5cd66065)

Closes https://github.com/ydb-platform/ydb-embedded-ui/issues/1210

need to redo https://github.com/ydb-platform/ydb-embedded-ui/pull/1198

### Test Status: <span style="color: orange;">⚠️ FLAKY</span>
📊 [Full Report](https://ydb-platform.github.io/ydb-embedded-ui/1256/)

| Total | Passed | Failed | Flaky | Skipped |
|:-----:|:------:|:------:|:-----:|:-------:|
| 124 | 117 | 0 | 7 | 0 |

### Bundle Size: ✅
Current: 78.86 MB | Main: 78.86 MB
Diff: 0.00 MB (-0.00%)

✅ Bundle size unchanged.

<details>
<summary>ℹ️ CI Information</summary>

- Test recordings for failed tests are available in the full report.
- Bundle size is measured for the entire 'dist' directory.
- 📊 indicates links to detailed reports.
- 🔺 indicates increase, 🔽 decrease, and ✅ no change in bundle size.
</details>

### CODE REVIEW

File: src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx [hunk 1]
```diff
@@ -466,35 +443,37 @@ function Result({
         return <Preview database={tenantName} path={path} type={type} />;
     }
 
-    if (resultType === RESULT_TYPES.EXECUTE) {
+    console.log(result);
+
```
Comment (ValeraS) [inline src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx:447]:
```suggestion
```

File: src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx [hunk 2]
```diff
@@ -113,18 +113,27 @@ function QueryEditor(props: QueryEditorProps) {
         LAST_USED_QUERY_ACTION_KEY,
     );
 
-    const [sendExecuteQuery, executeQueryResult] = executeQueryApi.useExecuteQueryMutation();
-    const [sendExplainQuery, explainQueryResult] = explainQueryApi.useExplainQueryMutation();
-    const [sendCancelQuery, cancelQueryResult] = cancelQueryApi.useCancelQueryMutation();
+    const [sendExecuteQuery] = executeQueryApi.useExecuteQueryMutation();
+    const [sendExplainQuery] = explainQueryApi.useExplainQueryMutation();
+    const [sendCancelQuery] = cancelQueryApi.useCancelQueryMutation();
```
Comment (ValeraS) [inline src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx:118]: Let's not put cancel query status in the store

```suggestion
    const [sendCancelQuery, cancelQueryState] = cancelQueryApi.useCancelQueryMutation();
    
    const queryId = executeQuery.result?.queryId;
    const cancelStatus =
        queryId && cancelQueryState.originalArgs?.queryId === queryId
            ? cancelQueryState.status
            : undefined;

```

Comment (ValeraS) [reply src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx:118]: Now `queryId` is in the state, so we can make a self contained `CancelButton` component.

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx (hunk 2 above)
Comment (astandrik) [reply src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx]: fixed

File: src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx [hunk 3]
```diff
@@ -96,12 +92,13 @@ function QueryEditor(props: QueryEditorProps) {
...
         theme,
         changeUserInput,
+        setQueryResult,
         showPreview,
     } = props;
     const {tenantPath: savedPath} = executeQuery;
 
-    const [resultType, setResultType] = React.useState<ValueOf<typeof RESULT_TYPES>>();
-    const [isResultLoaded, setIsResultLoaded] = React.useState(false);
+    const isResultLoaded = Boolean(executeQuery.result);
```
Comment (Raubzeug) [inline src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx:100]: why only `executeQuery.result`?

File: src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx [hunk 4]
```diff
@@ -335,7 +325,7 @@ function QueryEditor(props: QueryEditorProps) {
             <QueryEditorControls
                 handleSendExecuteClick={handleSendExecuteClick}
                 onSettingsButtonClick={handleSettingsClick}
-                isLoading={explainQueryResult.isLoading || executeQueryResult.isLoading}
+                isLoading={Boolean(executeQuery.result?.isLoading)}
```
Comment (Raubzeug) [inline src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx:328]: why only `executeQuery`?

File: src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx [hunk 5]
```diff
@@ -113,18 +110,26 @@ function QueryEditor(props: QueryEditorProps) {
...
         }
-    }, [changeUserInput, setPath, tenantName, savedPath]);
+    }, [
+        changeUserInput,
+        setPath,
+        setQueryResult,
+        tenantName,
+        savedPath,
+        executeQuery.result?.data,
+        executeQuery.result?.error,
```
Comment (ValeraS) [inline src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx:131]:
```suggestion
```

**Review by Raubzeug — CHANGES_REQUESTED** [review]

File: src/store/reducers/executeQuery.ts [hunk 6]
```diff
@@ -257,8 +277,24 @@ export const executeQueryApi = api.injectEndpoints({
...
                     dispatch(updateQueryInHistory(queryStats, queryId));
+                    dispatch(
+                        setQueryResult({
+                            type: ResultType.EXECUTE,
+                            data,
+                            isLoading: false,
+                            queryId,
+                        }),
+                    );
                     return {data};
```
Comment (ValeraS) [inline src/store/reducers/executeQuery.ts:288]: Let's not duplicate results and return empty data (`{data: null}`) here and in `explainQuery`.

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx [hunk 7]
```diff
@@ -375,22 +365,12 @@ function QueryEditor(props: QueryEditorProps) {
...
-                        executeResultLoading={executeQueryResult.isLoading}
-                        cancelQueryLoading={cancelQueryResult.isLoading}
                         resultVisibilityState={resultVisibilityState}
                         onExpandResultHandler={onExpandResultHandler}
                         onCollapseResultHandler={onCollapseResultHandler}
-                        onStopButtonClick={handleStopButtonClick}
                         type={type}
                         theme={theme}
-                        resultType={resultType}
+                        result={executeQuery.result}
```
Comment (ValeraS) [inline src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx:373]: Let's add a key to `Result`
```suggestion
                        key={executeQuery.result?.queryId}
                        result={executeQuery.result}
```

File: src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx (hunk 3 above)
Comment (astandrik) [reply src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx:100]: executeQuery.result can be both execute or explain type

File: src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx (hunk 4 above)
Comment (astandrik) [reply src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx:328]: executeQuery.result can be both execute or explain type

**Review by Raubzeug — APPROVED** [review]

**Review by ValeraS — APPROVED** [review]

**Pull request merged**